src/BizHawk.Client.Common/Api/MemoryMappedFiles.cs

```diff
@@ -8,33 +8,44 @@ namespace BizHawk.Client.Common
 	public sealed class MemoryMappedFiles
 	{
 		private readonly Dictionary<string, MemoryMappedFile> _mmfFiles = new Dictionary<string, MemoryMappedFile>();
-
 		private readonly Func<byte[]> _takeScreenshotCallback;
 
-		public string Filename;
+		public string? Filename;
 
-		public MemoryMappedFiles(Func<byte[]> takeScreenshotCallback, string filename)
+		public MemoryMappedFiles(Func<byte[]> takeScreenshotCallback, string? filename)
 		{
 			_takeScreenshotCallback = takeScreenshotCallback;
 			Filename = filename;
 		}
 
 		public string ReadFromFile(string filename, int expectedSize)
 		{
-			using var viewAccessor = MemoryMappedFile.OpenExisting(filename).CreateViewAccessor();
+			var bytes = ReadBytesFromFile(filename, expectedSize);
+			return Encoding.UTF8.GetString(bytes);
+		}
+		public byte[] ReadBytesFromFile(string filename, int expectedSize)
+		{
+			if (!_mmfFiles.TryGetValue(filename, out var mmfFile))
+			{
+				mmfFile = _mmfFiles[filename] = MemoryMappedFile.OpenExisting(filename);
+			}
+
+			var viewAccessor = mmfFile.CreateViewAccessor(0, expectedSize, MemoryMappedFileAccess.Read);
 			var bytes = new byte[expectedSize];
 			viewAccessor.ReadArray(0, bytes, 0, expectedSize);
-			return Encoding.UTF8.GetString(bytes);
+			viewAccessor.Dispose();
+			return bytes;
 		}
 
-		public int ScreenShotToFile() => WriteToFile(Filename, _takeScreenshotCallback());
+		public int ScreenShotToFile() => WriteToFile(Filename!, _takeScreenshotCallback());
```

Comment:
This should probably have better error handling:
```suggestion
		public int ScreenShotToFile()
		{
			if (Filename is null) {
				Console.WriteLine("MMF screenshot target not set; start EmuHawk with `--mmf=filename`");
				return 0;
			}
			return WriteToFile(Filename, _takeScreenshotCallback());
		}
```

And then the filename doesn't need to be checked in `CommLuaLibrary.MmfScreenshot`.